src/charm.py

```diff
@@ -130,6 +133,34 @@ def _on_get_admin_password(self, event: ActionEvent) -> None:
         credentials = jenkins.get_admin_credentials(self._jenkins_container)
         event.set_results({"password": credentials.password})
 
+    def _on_update_status(self, _: UpdateStatusEvent) -> None:
+        """Handle update status event.
+
+        On Update status:
+        1. Update Jenkins patch LTS version if available.
+        2. Update apt packages if available.
+        """
+        self.unit.status = ActiveStatus("Checking for updates.")
+
+        version = jenkins.get_version()
+        latest_patch_version = jenkins.get_latest_patch_version(version)
+        if latest_patch_version == version:
+            self.unit.status = ActiveStatus()
+            return
+
+        self.unit.status = MaintenanceStatus("Updating Jenkins.")
+        error = ""
+        try:
+            jenkins.download_stable_war(self._jenkins_container, latest_patch_version)
+            credentials = jenkins.get_admin_credentials(self._jenkins_container)
+            jenkins.safe_restart(credentials)
+        except (jenkins.JenkinsNetworkError, jenkins.JenkinsError) as exc:
+            error = "Failed to update Jenkins. See log for more details."
+            logger.error("Failed to update Jenkins. %s", exc)
+
+        self.unit.set_workload_version(latest_patch_version)
+        self.unit.status = ActiveStatus(error)
```

Comment:
My thought process was that even if there was an error updating Jenkins, as long as the service is still active, it doesn't quite categorize into other status.
For example, in the unlikely event of a corrupt RSS feed / executable download, it wouldn't affect the running Jenkins but a user's attention is required, hence the status message. This would be a nice case where a `warning` level of a status would fit in nicely but there currently isn't one.